### PR TITLE
First version of calendar integration with a camera

### PR DESCRIPTION
**Pull request by lukasz62**
Integration camera with a calendar. Made an example list desk_reservations in IoT_master/tmp_tab.py file - to be relaced later. I didn't run tox, I will do it later if the idea will be ok.

### CODE REVIEW

File: IoT_master/desks_reservation.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        #Start time and date of now - 10 minutes - time given for reservation
+        now = datetime.datetime.now()
+        now_10 = now - datetime.timedelta(minutes=10)
+        if(now_10 >= start_time and now <=end_time and desk_reservations[table_index] == 0):
+            #Checks if desk is reserved
+            #Delete an event if it is reserved but no one is sitting at the desk
+            CAL.events().delete(calendarId=result[2], eventId=event_id).execute()
+    page_token = events.get('nextPageToken')
+    if not page_token:
+        break
```
Comment (Dysproz) [inline IoT_master/desks_reservation.py]: Enclose it in:

```
if __name__ == '__main__':
```

File: IoT_master/tools/desks_reservation_func.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
+from __future__ import print_function
+import pickle
+import os.path
```
Comment (Dysproz) [inline IoT_master/tools/desks_reservation_func.py]: `from os import path`

File: IoT_master/tools/desks_reservation_func.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+def calendar_config():
+    '''
+    Open a configuration file and read iframe,
+    file with credentials for Calendar API and
+    calendarID
+    :return: List with iframe, api_credentials, calendarID strings
+    :rtype: List
+    '''
+    if(path.exists("../config.yml")):
+        with open('../config.yml', 'r') as file_config:
```
Comment (Dysproz) [inline IoT_master/tools/desks_reservation_func.py]: Usage of '../config.yml' may be a bit risky if function will be called in other context. If it should call file that's in directory above directory with this file then you can supersede dots with `dir_path = os.path.dirname(os.path.realpath(__file__))` which will give you always directory of your file.

File: IoT_master/tools/desks_reservation_func.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            content = yaml.load(file_config)
+        if 'calendar' in content:
+            iframe = content['calendar'].get('iframe', None)
+            api_credentials = content['calendar'].get('api_credentials', None)
+            calendarID = content['calendar'].get('calendarID', None)
+            desks = content['calendar'].get('desks', None)
```
Comment (Dysproz) [inline IoT_master/tools/desks_reservation_func.py:33]: .get() already returns None by default if there's no key

File: IoT_master/tools/desks_reservation_func.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+            api_credentials_path = os.path.join('../', api_credentials)
```
Comment (Dysproz) [inline IoT_master/tools/desks_reservation_func.py]: The same as above with dots

File: IoT_master/tools/desks_reservation_func.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    """Initialize a Google Calendar API
+    :return: Connection to Calendar
+    :rtype: googleapiclient.discovery.Resource
+    """
+    creds = None
+    # The file token.pickle stores the user's access and refresh tokens, and is
+    # created automatically when the authorization flow completes for the first
+    # time.
+    if os.path.exists('../token.pickle'):
+        with open('../token.pickle', 'rb') as token:
```
Comment (Dysproz) [inline IoT_master/tools/desks_reservation_func.py]: The same as above with dots

File: run.py [hunk 7]
```diff
@@ -382,23 +382,32 @@ def GUI_generate_service_principal(clicks):
...
-        end_date = datetime.datetime.strptime(end_date, '%Y-%m-%d').date()
-        if clicks != 0:
-            result = sendToGoogleCalendar(start_date, end_date, hours, minutes, title)
-        if result:
-            return True, False
-        else:
-            return False, True
+    if title is not None:
+        title_new = str(title).split()
+        if start_date is not None and end_date is not None:
```
Comment (Dysproz) [inline run.py]: Just
`if start_date and end_date:`

File: run.py [hunk 8]
```diff
@@ -382,23 +382,32 @@ def GUI_generate_service_principal(clicks):
...
+            start_date = datetime.datetime.strptime(start_date, '%Y-%m-%d').date()
+            end_date = datetime.datetime.strptime(end_date, '%Y-%m-%d').date()
+            if clicks != 0:
+                CAL = initializeCalendar()
+                print('After initialization')
+                if desk_available(CAL, title_new[1], desk_reservations) == False:
```
Comment (Dysproz) [inline run.py]: `if not  desk_available(CAL, title_new[1], desk_reservations)`

File: tools/calendar_generation.py [hunk 9]
```diff
@@ -53,7 +54,7 @@ def sendToGoogleCalendar(start_date, end_date, hours, minutes, event_title):
     :rtype: Boolean
     """
     try:
-        CAL = initializeCalendar()
+        #CAL = initializeCalendar()
```
Comment (Dysproz) [inline tools/calendar_generation.py]: If it's not necessary then delete it

**Review by Dysproz — CHANGES_REQUESTED** [review]
Overall, it's ok. I have some styling comments and also check flake8 command output

File: IoT_master/tools/desks_reservation_func.py (hunk 4 above)
Comment (lukasz62) [reply IoT_master/tools/desks_reservation_func.py:33]: So you suggest that I should remove 'None' in every get() function call?

Comment (Dysproz) [reply IoT_master/tools/desks_reservation_func.py:33]: Yes, because it’s already default

**Review by Dysproz — APPROVED** [review]
LGTM

**Pull request merged**